modules/apps/headless/headless-admin-site/headless-admin-site-impl/src/main/java/com/liferay/headless/admin/site/internal/resource/v1_0/util/LayoutUtil.java

```
@@ -111,6 +119,23 @@ public static Layout addContentLayout(
 				Boolean.FALSE.toString());
 		}
 
+		long masterLayoutPlid = 0L;
+
+		if ((settings != null) &&
+			Validator.isNotNull(
+				settings.getMasterPageItemExternalReference())) {
+
+			ItemExternalReference masterPageItemExternalReference =
+				settings.getMasterPageItemExternalReference();
+
+			Layout masterLayout =
+				LayoutLocalServiceUtil.getLayoutByExternalReferenceCode(
+					masterPageItemExternalReference.getExternalReferenceCode(),
```

Should we use fetch here and throw UnsupportedOperationException if null is returned, to indicate that we'll need to handle the reference in a lazy way?